cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -498,61 +498,166 @@ func (c *client) CreateMachines(machines []*clusterv1.Machine, namespace string)
 }
 
 func (c *client) DeleteClusters(namespace string) error {
-	err := c.clientSet.ClusterV1alpha1().Clusters(namespace).DeleteCollection(newDeleteOptions(), metav1.ListOptions{})
-	if err != nil {
-		return errors.Wrapf(err, "error deleting Clusters in namespace %q", namespace)
+	if namespace != "" {
+		err := c.clientSet.ClusterV1alpha1().Clusters(namespace).DeleteCollection(newDeleteOptions(), metav1.ListOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "error deleting Clusters in namespace %q", namespace)
+		}
+		err = c.waitForClusterDelete(namespace)
+		if err != nil {
+			return errors.Wrapf(err, "error waiting for Cluster(s) deletion to complete in namespace %q", namespace)
+		}
+		return nil
 	}
-	err = c.waitForClusterDelete(namespace)
+
+	seen := make(map[string]bool)
+	var namespaces []string
+	clusters, err := c.clientSet.ClusterV1alpha1().Clusters(namespace).List(metav1.ListOptions{})
```

Comment:
```suggestion
	clusters, err := c.clientSet.ClusterV1alpha1().Clusters("").List(metav1.ListOptions{})
```

Just to make it explicit that we're looking for clusters in all namespaces instead of using the empty variable